Sources/BrowserServicesKit/Autofill/WebsiteAutofillUserScript.swift

```diff
@@ -0,0 +1,135 @@
+        guard let websiteAutofillMessageName = WebsiteAutofillMessageName(rawValue: messageName) else {
+            return super.messageHandlerFor(messageName)
+        }
+        switch websiteAutofillMessageName {
+        case .getSelectedCredentials: return getSelectedCredentials
+        case .showAutofillParent: return showAutofillParent
+        case .closeAutofillParent: return closeAutofillParent
+        }
+    }
+
+    func showAutofillParent(_ message: AutofillMessage, _ replyHandler: MessageReplyHandler) {
+        guard let dict = message.messageBody as? [String: Any],
+              let left = dict["inputLeft"] as? CGFloat,
+              let top = dict["inputTop"] as? CGFloat,
+              let height = dict["inputHeight"] as? CGFloat,
+              let width = dict["inputWidth"] as? CGFloat,
+              let serializedInputContext = dict["serializedInputContext"] as? String,
+              let currentOverlayTab = currentOverlayTab,
+              let clickPoint = clickPoint else {
+                  return
+              }
+        // Sets the last message host, so we can check when it messages back
+        lastOpenHost = hostProvider.hostForMessage(message)
+
+        currentOverlayTab.websiteAutofillUserScript(self,
+                                                 willDisplayOverlayAtClick: clickPoint,
+                                                serializedInputContext: serializedInputContext,
+                                                 inputPosition: CGRect(x: left, y: top, width: width, height: height))
+        replyHandler(nil)
+    }
+
+    func closeAutofillParent(_ message: AutofillMessage, _ replyHandler: MessageReplyHandler) {
+        close()
+        replyHandler(nil)
+    }
+
+    internal func close() {
+        guard let currentOverlayTab = currentOverlayTab else { return }
+        currentOverlayTab.websiteAutofillUserScriptCloseOverlay(self)
+        selectedDetailsData = nil
+        lastOpenHost = nil
+    }
+
+    /// Called from the child autofill to return referenced credentials
+    func getSelectedCredentials(_ message: AutofillMessage, _ replyHandler: MessageReplyHandler) {
+        var response = GetSelectedCredentialsResponse(type: "none")
+        if lastOpenHost == nil || message.messageHost != lastOpenHost {
+            response = GetSelectedCredentialsResponse(type: "stop")
+        } else if let selectedDetailsData = selectedDetailsData {
+            response = GetSelectedCredentialsResponse(type: "ok", data: selectedDetailsData.data, configType: selectedDetailsData.configType)
+            self.selectedDetailsData = nil
+        }
+        if let json = try? JSONEncoder().encode(response),
+           let jsonString = String(data: json, encoding: .utf8) {
+            replyHandler(jsonString)
+        }
+    }
+}
+
+
+/// Is used by the top Autofill to reference into the child autofill
+public protocol AutofillMessagingToChildDelegate {
```

Comment:
Maybe let's call it `OverlayAutofillUserScriptDelegate` and move it to OverlayAutofillUserScript file?